### PR TITLE
wifi: MBO Pre-certification bugfixes

### PR DESCRIPTION
**Pull request by sr1dh48r**
Support for handling MLME frames.
Enable RRM support in capabilities.
Support for wnm_bss_query wpa_cli cmd.

Signed-off-by: Sridhar Nuvusetty <sridhar.nuvusetty@nordicsemi.no>

### CODE REVIEW

File: drivers/wifi/nrf700x/zephyr/src/zephyr_wpa_supp_if.c [hunk 1]
```diff
@@ -1016,3 +1016,89 @@ void wifi_nrf_wpa_supp_event_proc_get_if(void *if_priv,
...
+	if (data_len) {
+		memcpy(mgmt_tx_info->frame.frame, data, data_len);
+		mgmt_tx_info->frame.frame_len = data_len;
+	}
+
+	mgmt_tx_info->freq_params.frequency = freq;
+	mgmt_tx_info->freq_params.channel_width = NRF_WIFI_CHAN_WIDTH_20;
+	mgmt_tx_info->freq_params.center_frequency1 = freq;
+	mgmt_tx_info->freq_params.center_frequency2 = 0;
+	mgmt_tx_info->freq_params.channel_type = NRF_WIFI_CHAN_WIDTH_20;
```
Comment (sachinthegreen) [inline drivers/wifi/nrf700x/zephyr/src/zephyr_wpa_supp_if.c:1085]: Is NRF_WIFI_CHAN_WIDTH_20 the correct macro to be used here (i.e. should there be a different macro for channel type)?

File: drivers/wifi/nrf700x/zephyr/src/zephyr_wpa_supp_if.c [hunk 2]
```diff
@@ -1016,3 +1016,89 @@ void wifi_nrf_wpa_supp_event_proc_get_if(void *if_priv,
 
 	k_sem_give(&wait_for_event_sem);
 }
+
+void wifi_nrf_wpa_supp_event_mgmt_tx_status(void *if_priv,
+		struct nrf_wifi_umac_event_mlme *mlme_event,
+		unsigned int event_len)
+{
+	struct wifi_nrf_vif_ctx_zep *vif_ctx_zep = NULL;
+
```
Comment (sachinthegreen) [inline drivers/wifi/nrf700x/zephyr/src/zephyr_wpa_supp_if.c:1025]: Good to have a defensive check here to see that if_priv and mlme_event are not NULL.

**Review by sachinthegreen — CHANGES_REQUESTED** [review]

**Comment by sr1dh48r** [comment]
@krish2718 @sachinthegreen reminder on this !! Ajay is waiting on this PR to make further changes ..

**Review by krish2718 — APPROVED** [review]
LGTM, but you need to rebase for CI to pass.

**Review by sachinthegreen — APPROVED** [review]

**Pull request merged**